lib/backend/etcd/etcd.go

```diff
@@ -92,21 +103,18 @@ func ConnectEtcdClient(config *api.ClientConfig) (DatastoreReadWriteInterface, e
 }
 
 // Create an entry in the datastore.  This errors if the entry already exists.
-func (c *EtcdClient) Create(d *DatastoreObject) (*DatastoreObject, error) {
-	// Special case the profile
-	switch t := d.Key.(type) {
-	case ProfileKey:
-		return t.Create(c, d)
+func (c *EtcdClient) Create(d *KVPair) (*KVPair, error) {
+	if co, ok := d.Key.(CreateOverrider); ok {
```

Comment:
Perhaps we should add the override methods to the Profile object instead of the Key... although I had a sneaky feeling there was a reason I didn't like that - but I can't think of it now.